gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/CodeableConceptMapper.java

```diff
@@ -40,231 +39,274 @@ public CodeableConcept mapToCodeableConceptForMedication(CD codedData) {
+    private String getTextFieldValue(String text, String newText) {
+        return StringUtils.isNotEmpty(text)
+            ? text
+            : newText;
     }
-
+    
     private boolean isCodeConceptId(String partitionIdentifier) {
-        return CONCEPT_PARTITION_SHORT.equals(partitionIdentifier) || CONCEPT_PARTITION_LONG.equals(partitionIdentifier);
+        return CONCEPT_PARTITION_SHORT.equals(partitionIdentifier)
+            || CONCEPT_PARTITION_LONG.equals(partitionIdentifier);
     }
 
     private boolean isCodeDescriptionId(String partitionIdentifier) {
-        return DESCRIPTION_PARTITION_SHORT.equals(partitionIdentifier) || DESCRIPTION_PARTITION_LONG.equals(partitionIdentifier);
+        return DESCRIPTION_PARTITION_SHORT.equals(partitionIdentifier)
+            || DESCRIPTION_PARTITION_LONG.equals(partitionIdentifier);
     }
 
-    private boolean preferredValuePresentNoDescriptionValue(SnomedCTDescription preferredTerm, SnomedCTDescription description) {
-        return Objects.nonNull(preferredTerm) && Objects.isNull(description);
+    private boolean hasPreferredValuePresentAndNoDescriptionValue(SnomedCTDescription preferredTerm, SnomedCTDescription description) {
+        return Objects.nonNull(preferredTerm)
+            && Objects.isNull(description);
     }
 
-    private boolean preferredAndDescriptionValuesPresent(SnomedCTDescription preferredTerm, SnomedCTDescription description) {
-        return Objects.nonNull(preferredTerm) && Objects.nonNull(description);
+    private boolean hasPreferredValueAndDescriptionValuePresent(SnomedCTDescription preferredTerm, SnomedCTDescription description) {
+        return Objects.nonNull(preferredTerm)
+            && Objects.nonNull(description);
     }
 
-    private boolean preferredAndDescriptionValuesNotPresent(SnomedCTDescription preferredTerm, SnomedCTDescription description) {
-        return Objects.isNull(preferredTerm) && Objects.isNull(description);
+    private boolean hasPreferredValueAndDescriptionValueNotPresent(SnomedCTDescription preferredTerm, SnomedCTDescription description) {
+        return Objects.isNull(preferredTerm)
+            && Objects.isNull(description);
     }
 
-    private boolean translationMainCodeHasOriginalText(CD mainCodeFromTranslation) {
-        return Objects.nonNull(mainCodeFromTranslation) && StringUtils.isNotEmpty(mainCodeFromTranslation.getOriginalText());
+    private boolean hasOriginalTextInTranslationMainCode(CD translationMainCode) {
+        return Objects.nonNull(translationMainCode)
+            && StringUtils.isNotEmpty(translationMainCode.getOriginalText());
     }
 
-    private boolean translationMainCodeHasDisplayName(CD mainCodeFromTranslation) {
-        return Objects.nonNull(mainCodeFromTranslation) && StringUtils.isNotEmpty(mainCodeFromTranslation.getDisplayName());
+    private boolean hasDisplayNameInTranslationMainCode(CD mainCodeFromTranslation) {
```

Comment:
❤️ putting the `has` at the beginning of a method name, makes it so much clearer.

Reply:
yeah in my opinion (except for special cases!) any method which returns a boolean should start an auxiliary verb.